File: Ex44_RandomNumberGen.java

public class Ex44_RandomNumberGen{
	public static void main(String[] args){
		int i = 1;
		int x = 0;
		int y = 1;
		int found1 = 0;
		int found2 = 0;
		int both = 0;
		int hold1 = 0;
		int hold2 = 0;
		
		while(y == 1){
			
			x = (int)(Math.random()*(-19) + Math.random()*10);
			if (x == 9){
			    System.out.println("Found 9 at " + i);
				if(found1 == 0){
					hold1 = i;
				}
				found1 = 1; // to indicate first found 9
				
			    if (found2 == 1){
					both = 1; // both found
			    }
			}
			if (x == (-18)){
			    System.out.println("Found -18 at " + i);
				if(found2 == 0){
					hold2 = i;
				}
				found2 = 1; // to indicate first found 9
				
			    if (found1 == 1){
					both = 1; // both found
			    }
			}
			if (both == 1){
				break;
			}
			
			i = i + 1;
		} // end while
		
		System.out.println("The first appearance of -18 was at " + hold2);
		System.out.println("The first appearance of 9 was at " + hold1);
		
	}
}

File: Ex44_RandomNumberGen.c
#include <stdio.h>
#include <time.h>

int main(){
	
	srand(time(0));
	int i, x, both, hold1, hold2;
	int found1 = 0; int found2 = 0;
	
	for(i = 0; ;i++){
		
		x = (rand()%10)-(rand()%19);
		
		if(x==9){
			printf("Found 9 at %d\n",i);
			if(found1 == 0){
				hold1 = i;
			}
			found1 = 1; /* 9 has been found */
			
			if(found2 == 1){
				both = 1; /* both has been found */
			}
		}
		if(x==(-18)){
			printf("Found -18 at %d\n",i);
			if(found2 == 0){
				hold2 = i;
			}
			found2 = 1; /* -18 has been found */
			
			if(found1 == 1){
				both = 1; /* both has been found */
			}
		}
		if(both==1){
			break;
		}
	} // end for
	
	printf("The first appearance of -18 was at %d\n", hold2);
	printf("The first appearance of 9 was at %d\n", hold1);
	
	return 0;
}
